lib/Service/DocumentService.php

```diff
@@ -125,11 +124,11 @@ public function createDocument(File $file): Document {
 
 			// Do not hard reset if changed from outside since this will throw away possible steps
 			// This way the user can still resolve conflicts in the editor view
-			if ($document->getLastSavedVersion() !== $document->getCurrentVersion()) {
+			$stepsVersion = $this->stepMapper->getLatestVersion($document->getId());
+			if ($stepsVersion && ($document->getLastSavedVersion() !== $stepsVersion)) {
```

Comment:
If we drop the step version that check against the last saved version will no longer work, so maybe we need to find a different approach to determine the collision case.